src/main/scala/io/iohk/ethereum/common/SimpleMap.scala

```diff
@@ -0,0 +1,63 @@
+package io.iohk.ethereum.common
+
+/**
+  * Interface to represent a key-value structure
+  */
+trait SimpleMap[K, V] {
+
+  type T <: SimpleMap[K, V]
+
+  /**
+    * This function obtains the value asociated with the key passed, if there exists one.
+    *
+    * @param key
+    * @return Option object with value if there exists one.
+    */
+  def get(key: K): Option[V]
+
+  /**
+    * This function inserts a (key-value) pair into the trie. If the key is already asociated with another value it is updated.
+    *
+    * @param key
+    * @param value
+    * @return New trie with the (key-value) pair inserted.
+    */
+  def put(key: K, value: V): T = update(Seq(), Seq(key -> value))
+
+  /**
+    * This function inserts a (key-value) pair into the trie. If the key is already asociated with another value it is updated.
+    *
+    * @param kv to insert
+    * @return New trie with the (key-value) pair inserted.
+    */
+  def +(kv: (K, V)): T = put(kv._1, kv._2)
+
+
+  /**
+    * This function deletes a (key-value) pair from the trie. If no (key-value) pair exists with the passed trie then there's no effect on it.
+    *
+    * @param key
+    * @return New trie with the (key-value) pair associated with the key passed deleted from the trie.
+    */
+  def remove(key: K): T = update(Seq(key), Seq())
```

Comment:
What if I would like to put empty value with some key, or there will not be such case?